docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+                                );
+   
+   // Register FileEndpoint as "SALES" table with file path and data format as csv
+   sqlEnv = sqlEnv.registerTable( 
+                                  "SALES", 
+                                  new FileEndpoint("/tmp/output", 
+                                                   "out.file", 
+                                                   new CSVMessageFormat(<>))
+                                );
+                                
+   // Register scalar SQL UDF 
+   sqlEnv = sqlEnv.registerFunction("APEXCONCAT", PureStyleSQLApplication.class, "apex_concat_str");
+   
+   // Converting SQL statement to DAG 
+   String sql = "INSERT INTO SALES 
+                     SELECT STREAM ROWTIME, FLOOR(ROWTIME TO DAY), APEXCONCAT('OILPAINT', SUBSTRING(PRODUCT, 6, 7)) 
+                     FROM ORDERS 
+                     WHERE ID > 3 AND PRODUCT LIKE 'paint%'";
+   sqlEnv.executeSQL(dag, sql);
+ }// populateDAG finished
+ 
+ public static String apex_concat_str(String s1, String s2)
+ {
+      return s1 + s2;
+ } 
+}
+```      
+### Constructing SQLExecEnvironment
+Class _**SQLExecEnvironment**_ provides a starting point and a simple way to define metadata needed for running SQL statement. SQLExecEnvironment.getEnvironment() creates a new instance of SQLExecEnvironment.  
+
+```java
+// Creates SQLExecEnvironment instance by using static method getEnvironment
+SQLExecEnvironment sqlEnv = SQLExecEnvironment.getEnvironment();
+```
+
+### Registering tables with SQLExecEnvironment
+Next, we need to register tables which can be used in a query. For this purpose, we can use _registerTable_ API from SQLExecEnvironment.
+```java
+// Register KafkaEnpoint as "ORDERS" table with kafka topic and data format as CSV
+sqlEnv = sqlEnv.registerTable( 
+                              "ORDERS", 
+                              new KafkaEndpoint("localhost:9090", 
+                                                "inputTopic", 
+                                                new CSVMessageFormat(<>))
+                            );
+
+// Register FileEndpoint as "SALES" table with file path and data format as csv
+sqlEnv = sqlEnv.registerTable( 
+                              "SALES", 
+                              new FileEndpoint("/tmp/output", 
+                                               "out.file", 
+                                               new CSVMessageFormat(<>))
+                            );
+```
+
+**_"registerTable"_** API takes the name of the table and an instance of endpoint as parameters. Endpoint signifies data storage mechanism and type of source/destination for the data. Each of these endpoints requires a different type of configurations and possibly data formats. In this example, data format is defined using _MessageFormat_ by specifying it at the endpoint creation. _CSVMessageFormat_ can be configured with schema string as follows:
```

Comment:
Each of these endpoints requires a different type of => These endpoints require different types of
data format is defined using _MessageFormat_ .... as follows: => the data format is defined using an implementation of the _MessageFormat_ interface; the _CSVMessageFormat_ implementation can be configured with a schema string as follows: